src/Config/FileLocatorFactoryInterface.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Config;
+
+use Symfony\Component\Config\FileLocatorInterface;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Interface for file locator factories.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+interface FileLocatorFactoryInterface
```

Comment:
Yes, but they are not static classes, at least quick scanning did not find any.
Can you please show me the static (only!) factories in Symfony implementing an interface? :)

Until then I explain that it feels plain wrong to implement it like this and show counter examples.

First example: [```php class FormFactory implements FormFactoryInterface```](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/Forms.php#L24)
```php
use Symfony\Component\Form\Forms;
// The static call creates the factory instead of _BEING_ the factory.
$formFactory = Forms::createFormFactory(); 
$form = $formFactory->createBuilder()
    ->add('firstName', 'text')
    ->add('lastName', 'text')
    ->add('age', 'integer')
    ->add('gender', 'choice', array(
        'choices' => array('m' => 'Male', 'f' => 'Female'),
    ))
    ->getForm();
```
It is perfectly valid to have an interface here, as you get a factory instance which will perform the work.
You do NOT have any object here performing work but merely have a bunch of static functions stuck together into a "namespace" (class) which feels plain wrong.

Next example: [MetadataFactoryInterface](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Mapping/Factory/MetadataFactoryInterface.php) implemented by [BlackHoleMetadataFactory](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Mapping/Factory/BlackHoleMetadataFactory.php) and [LazyLoadingMetadataFactory](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php) is also not static (You might have to look at the deprecated interface until it get's moved to the new location in 3.0 over [here](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/MetadataFactoryInterface.php)).

You need the interface when implementing the "abstract factory pattern", not when the factory class is the "final" implementation.